shell/components/ResourceList/index.vue

```diff
@@ -67,6 +67,11 @@ export default {
         this.loadResources = loadResources || [resource];
         this.loadIndeterminate = loadIndeterminate || false;
       }
+
+      // If the custom component contains the paginated resource table it'll control the fetching
+      if (component?.components?.['PaginatedResourceTable']) {
+        this.componentWillFetch = true;
+      }
```

Comment:
What is this resolving? This looks like a gotcha waiting to happen, I also don't like how we're adding more implicit logic which will cause more effort when we try to de-tangle and make things more composable.